hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/TokenContextImpl.java

```diff
@@ -143,4 +149,11 @@ public boolean isScheduleDispatch() {
     public void markMigrationRecordsStreamed() {
         blockRecordManager.markMigrationRecordsStreamed();
     }
+
+    @Override
+    public Set<Long> knownNodeIds() {
+        return new ReadableStoreFactory(state)
```

Comment:
I think we can use `stack` here. Both will return the same `Set`, but I think using the root state and stack in the same class risks the introduction of hard-to-spot bugs in the future.